uplink/src/base/serializer/mod.rs

```diff
@@ -134,21 +136,23 @@ struct StorageHandler {
 impl StorageHandler {
     fn new(config: Arc<Config>) -> Result<Self, Error> {
         let mut map = HashMap::with_capacity(2 * config.streams.len());
-        for (stream_name, config) in config.streams.iter() {
-            let mut storage = Storage::new(config.persistence.max_file_size);
-            if let Some(persistence) = &config.persistence.disk {
-                let mut path = persistence.path.clone();
+        for (stream_name, stream_config) in config.streams.iter() {
+            let mut storage = Storage::new(stream_config.persistence.max_file_size);
+            if stream_config.persistence.max_file_count > 0 {
+                let mut path = config.persistence_path.clone();
                 path.push(stream_name);
 
+                std::fs::create_dir_all(&path).map_err(|_| {
```

Comment:
Check if uplink is exiting with good error message during this case

Reply:
Create `negative cases` section in QA doc and add

- Unable to access persistence path
- Unable to create a dir
- Config file not available 
- Auth file not available
- Config fiile/Auth file is not valid json or toml